File: my_py_pkg/my_py_pkg/number_counter.py
#!usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import Int64
from example_interfaces.srv import SetBool

class number_counter(Node):
    def __init__(self):
        super().__init__("number_couter")
        self.subscriber_ = self.create_subscription(Int64, "number", self.callback_counter, 10)
        self.publisher_ = self.create_publisher(Int64, "number_count", 10)
        self.server_ = self.create_service(SetBool, "reset_counter", self.callback_reset_counter)
        self.total_count = 0
        self.get_logger().info("Number counter has been started.")

    def callback_counter(self, msg: Int64):
        self.total_count += msg.data
        self.get_logger().info(f"data: {self.total_count}")
        total_msg = Int64()
        total_msg.data = self.total_count
        self.publisher_.publish(total_msg)

    def callback_reset_counter(self, request, response):
        if request.data:
            self.get_logger().info("Reseting counter")
            self.total_count = 0
            response.success = True
            response.message = "reset finished"
            return response
        else:
            response.success = False
            response.message = "counter still going"
        return response

def main(args=None):
    rclpy.init(args=args)
    node = number_counter()
    rclpy.spin(node)
    rclpy.shutdown()
if __name__ == "__main__":
    main()

File: my_cpp_pkg/src/number_counter.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/int64.hpp"
using namespace std::chrono_literals;
using namespace std::placeholders;
class NumberCounter : public rclcpp::Node
{
public:
    NumberCounter(): Node("number_counter"), total_count_(0)
    {
        publisher_ = create_publisher<example_interfaces::msg::Int64>("number_count", 10);
        subscription_ = create_subscription<example_interfaces::msg::Int64>("number", 10, std::bind(&NumberCounter::numbering_callback, this, _1));
        RCLCPP_INFO(get_logger(), "Numbering Started");
    }
private:
    void numbering_callback(const example_interfaces::msg::Int64::SharedPtr msg)
    {
        total_count_ += msg->data;
        RCLCPP_INFO(get_logger(), "received number: %ld", total_count_);
        auto total_msg = example_interfaces::msg::Int64();
        total_msg.data = total_count_;
        publisher_->publish(total_msg);
    }
    rclcpp::Publisher<example_interfaces::msg::Int64>::SharedPtr publisher_;
    rclcpp::Subscription<example_interfaces::msg::Int64>::SharedPtr subscription_;
    int64_t total_count_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<NumberCounter>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}